### PR TITLE
drivers: nxp: convert SIUL2 drivers to native

### PR DESCRIPTION
**Pull request by manuargue**
Convert pin control, GPIO and external interrupt controller drivers based on SIUL2 peripheral to native drivers.
Tested with `west twister --tag gpio` for `s32z2xxdc2@D/s32z270/rtu0` and `mr_canhubk3` boards.

> [!NOTE]  
> Changes must be kept in a single commit to preserve atomicity, as these drivers depend on each other.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]
:100:

**Review by bperseghetti — DISMISSED** [review]

**Review by PetervdPerk-NXP — DISMISSED** [review]

File: drivers/gpio/gpio_nxp_s32.c [hunk 1]
```diff
@@ -66,13 +75,24 @@ struct gpio_nxp_s32_data {
 #endif
 };
 
+static inline uint16_t reverse_bits_16(uint16_t value)
```
Comment (PetervdPerk-NXP) [inline drivers/gpio/gpio_nxp_s32.c:78]: Not sure if zephyr exposes inline ARM assembly but you could swap this with
https://developer.arm.com/documentation/ddi0403/d/Application-Level-Architecture/Instruction-Details/Alphabetical-list-of-ARMv7-M-Thumb-instructions/RBIT?lang=en

**Review by dleach02 — DISMISSED** [review]

File: drivers/gpio/gpio_nxp_s32.c (hunk 1 above)
Comment (manuargue) [reply drivers/gpio/gpio_nxp_s32.c]: changed to `__RBIT` from cmsis

**Review by Dat-NguyenDuy — DISMISSED** [review]

**Review by PetervdPerk-NXP — DISMISSED** [review]

**Review by bperseghetti — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]
CI is failing, but seems is not caused by the changes in this PR

**Comment by manuargue** [comment]
> CI is failing, but seems is not caused by the changes in this PR

will be fixed in https://github.com/zephyrproject-rtos/zephyr/pull/76449

**Pull request merged**